packages/ruleset/src/functions/schema-casing-convention.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2023 IBM Corporation.
+ * SPDX-License-Identifier: Apache2.0
+ */
+
+const { isObject } = require('@ibm-cloud/openapi-ruleset-utilities');
+const { casing } = require('@stoplight/spectral-functions');
+const { LoggerFactory } = require('../utils');
+
+let casingConfig;
+let ruleId;
+let logger;
+
+module.exports = function (components, options, context) {
+  // Save this rule's "functionOptions" value since we need
+  // to pass it on to Spectral's "casing" function.
+  casingConfig = options;
+
+  if (!logger) {
+    ruleId = context.rule.name;
+    logger = LoggerFactory.getInstance().getLogger(ruleId);
+  }
+
+  return schemaCaseConvention(components, context.path);
+};
+
+function schemaCaseConvention(components, path) {
+  if (!components.schemas && !isObject(components.schemas)) {
```

Comment:
```suggestion
  if (!components.schemas || !isObject(components.schemas)) {
```
Should this be an OR instead of an AND?

Reply:
I don't know why I keep doing that 😅 I will fix!